File: library/src/main/java/su/levenetc/android/textsurface/interfaces/ISurfaceAnimation.java
package su.levenetc.android.textsurface.interfaces;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import su.levenetc.android.textsurface.TextSurface;

/**
 * Created by Eugene Levenetc.
 */
public interface ISurfaceAnimation {

	void onStart();

	void start(@Nullable IEndListener listener);

	void setTextSurface(@NonNull TextSurface textSurface);

	long getDuration();

	void cancel();

}

File: library/src/main/java/su/levenetc/android/textsurface/interfaces/ISurfaceAnimation.kt
package su.levenetc.android.textsurface.interfaces

import su.levenetc.android.textsurface.TextSurface

/**
 * Created by Eugene Levenetc.
 */
interface ISurfaceAnimation {

    val duration: Long
    var textSurface: TextSurface

    fun onStart()
    fun start(listener: IEndListener)
    fun cancel()
}